src/ddmd/expressionsem.d

```diff
@@ -4769,30 +4417,25 @@ extern (C++) final class ExpressionSemanticVisitor : Visitor
             return;
         }
         if (exp.e1.type.ty == Terror)
-        {
-            result = new ErrorExp();
-            return;
-        }
+            return setError();
 
         Type t1b = exp.e1.type.toBasetype();
         if (t1b.ty == Tpointer)
         {
             if ((cast(TypePointer)t1b).next.ty == Tfunction)
             {
                 exp.error("cannot slice function pointer %s", exp.e1.toChars());
-                result = new ErrorExp();
+                return setError();
```

Comment:
There was no `return` here before, but probably a bug.
(not sure my first comment got trough, I received an error from github.)

Reply:
Just checked the commit, it is indeed a bug introduced by the conversion to visitor.